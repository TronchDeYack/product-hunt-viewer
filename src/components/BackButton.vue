<template>
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 64 64" @click="goBack">
    <g clip-path="url(#b)">
      <path d="M48.031 30.031v3.938H23.656l11.156 11.25L32
      48.031 15.969 32 32 15.969l2.812 2.812-11.156 11.25z"/>
    </g>
  </svg>
</template>

<style lang="scss" scoped>
svg {
  width: 30px;

  &:hover {
    cursor: pointer;
  }
}
</style>

<script>
export default {
  name: 'back-button',
  methods: {
    goBack() {
      this.$router.go(-1);
    },
  },
};
</script>
