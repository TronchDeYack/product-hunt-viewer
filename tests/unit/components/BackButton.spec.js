import { mount } from '@vue/test-utils';

import BackButton from '@/components/BackButton.vue';

describe('BackButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(BackButton);
  });

  it('Should render.', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should back to previous page when clicking on it.', () => {
    wrapper.find('svg').trigger('click');

    expect(wrapper.vm.$router.go).toHaveBeenCalledWith(-1);
  });
});
