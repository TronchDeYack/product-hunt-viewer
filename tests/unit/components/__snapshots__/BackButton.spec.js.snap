// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Should render. 1`] = `
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 64 64">
  <g clip-path="url(#b)">
    <path d="M48.031 30.031v3.938H23.656l11.156 11.25L32
    48.031 15.969 32 32 15.969l2.812 2.812-11.156 11.25z"></path>
  </g>
</svg>
`;
